ansible_collections/arista/avd/roles/eos_designs/python_modules/interface_descriptions/avdinterfacedescriptions.py

```diff
@@ -43,8 +43,18 @@ def underlay_ethernet_interface(self, data: InterfaceDescriptionData) -> str:
             - overlay_routing_protocol
             - type
             - vrf
+            - wan_carrier
+            - wan_circuit_id
         """
-        desc = self.underlay_ethernet_interfaces(link_type=data.link_type, link_peer=data.peer, link_peer_interface=data.peer_interface)
+        if data.wan_carrier:
```

Comment:
```suggestion
        if data.wan_carrier or data.wan_circuit_id:
```

Reply:
can wan circuit id be present without wan carrier ?

Reply:
Well there is nothing in the logic before this step that hinders that, so we should not infer such a restriction this deep in the chain.

Reply:
Comment no longer applicable, changed code to fix the bug you pointed. 

The bug was not due to custom interface descriptions module, but rather due to `if data.wan_carrier` check, as for l3_interfaces that didn't have wan_carrier we were not emitting anything.